### PR TITLE
test: Fixed e2e test to work in local and ci

### LINKED ISSUE #303: Unable to run e2e test against public registry
### Please fill out the issue checklist below and provide ALL the requested information.

- [x] I reviewed open and closed Github issues that may be related to my problem.
- [x] I am reporting a bug that others will be able to reproduce.

### Describe the bug
Run e2e test against public registry like GCR

### Expected behavior
Expected to run e2e test against public registry like GCR. This makes development loop reproduce able. 

### Steps to Reproduce
`BUNDLE=gcr.io/some-project/test-integration:latest make integration-test`

### PR DESCRIPTION
**Pull request by odinnordico**
# Pull request

### What this PR does / why we need it

Changed cluster supply chain e2e so it runs for all clusters
Add skip to e2e tests if not required Env Vars are set
Added local private registry in ci for ca cert tests

### Which issue(s) this PR fixes

Fixes #303 

### Describe testing done for PR

local test with `make integration-test` and ci test in fork: https://github.com/odinnordico/apps-cli-plugin/actions/runs/2965680066

### CODE REVIEW

**Comment by odinnordico** [comment]
Some lines and format changes were generated by `go fmt ./...`

File: testing/e2e/cluster_supply_chain_test.go [hunk 1]
```diff
@@ -21,22 +21,56 @@ package integration_test
...
-			ExpectedCommandLineOutput: it.GetFileAsString(t, filepath.Join(it.ConsoleOutBasePath, "list-csc", "test-list-csc.txt")),
+			Name:    "List the existing supply chains",
+			Command: *it.NewTanzuAppsCommandLine("cluster-supply-chain", "list"),
+			Verify: func(t *testing.T, output string, err error) {
+				match, err := regexp.MatchString(`.*.\nci-test[ ]{3,}[<>a-zA-Z]{5,}[ ]{3,}[0-9]{1,}s\n.*`, output)
+				if !match {
+					t.Error("Expected 'ci-test   <status>   0s' to be present in the output")
+					t.FailNow()
+				}
+				if err != nil {
```
Comment (shashwathi) [inline testing/e2e/cluster_supply_chain_test.go:43]: check for error first and then the match variable.

File: testing/e2e/cluster_supply_chain_test.go [hunk 2]
```diff
@@ -21,22 +21,56 @@ package integration_test
...
+					t.Error("Error while validating the output", err)
+					t.FailNow()
+				}
+				const outFoot = `
```
Comment (shashwathi) [inline testing/e2e/cluster_supply_chain_test.go:47]: update name to `expectedFooter`  and make it var.

File: testing/e2e/cluster_supply_chain_test.go [hunk 3]
```diff
@@ -21,22 +21,56 @@ package integration_test
...
+			},
 		},
 		{
-			Name:                      "Get the existing supply chain",
-			Command:                   *it.NewTanzuAppsCommandLine("cluster-supply-chain", "get", "ci-test"),
-			ExpectedCommandLineOutput: it.GetFileAsString(t, filepath.Join(it.ConsoleOutBasePath, "get-csc", "test-get-ci-test-csc.txt")),
+			Name:    "Get the existing supply chain",
+			Command: *it.NewTanzuAppsCommandLine("cluster-supply-chain", "get", "ci-test"),
+			Verify: func(t *testing.T, output string, err error) {
+				const outHead = "---\n# ci-test:"
```
Comment (shashwathi) [inline testing/e2e/cluster_supply_chain_test.go:62]: update name from `outHead` to `expectedHeader`

File: testing/e2e/cluster_supply_chain_test.go [hunk 4]
```diff
@@ -21,22 +21,56 @@ package integration_test
...
+				if !strings.HasPrefix(output, outHead) {
+					t.Errorf("Expected %s to be present in the output", outHead)
+					t.FailNow()
+				}
+				valContent := it.GetFileAsString(t, filepath.Join(it.ConsoleOutBasePath, "get-csc", "test-get-ci-test-csc.txt"))
+				if diff := cmp.Diff(valContent, output[len(output)-len(valContent):]); diff != "" {
```
Comment (shashwathi) [inline testing/e2e/cluster_supply_chain_test.go:68]: Update the diff to use regex like in other test. Expect the supply chain Table heading row and an entry for the ci-test row as well

**Review by shashwathi — CHANGES_REQUESTED** [review]

File: testing/e2e/cluster_supply_chain_test.go (hunk 1 above)
Comment (odinnordico) [reply testing/e2e/cluster_supply_chain_test.go:37]: Done

File: testing/e2e/cluster_supply_chain_test.go (hunk 2 above)
Comment (odinnordico) [reply testing/e2e/cluster_supply_chain_test.go]: Updated

File: testing/e2e/cluster_supply_chain_test.go (hunk 3 above)
Comment (odinnordico) [reply testing/e2e/cluster_supply_chain_test.go]: Updated

File: testing/e2e/cluster_supply_chain_test.go (hunk 4 above)
Comment (odinnordico) [reply testing/e2e/cluster_supply_chain_test.go]: changed to use regex

File: testing/e2e/cluster_supply_chain_test.go [hunk 5]
```diff
@@ -29,14 +30,49 @@ import (
...
+			Name:    "List the existing supply chains",
+			Command: *it.NewTanzuAppsCommandLine("cluster-supply-chain", "list"),
+			Verify: func(t *testing.T, output string, err error) {
+				match, err := regexp.MatchString(`.*.\nci-test[ ]{3,}[<>a-zA-Z]{5,}[ ]{3,}[0-9]{1,}s\n.*`, output)
+				if err != nil {
+					t.Error("Error while validating the output", err)
+					t.FailNow()
+				}
+				if !match {
+					t.Error("Expected 'ci-test   <status>   0s' to be present in the output")
```
Comment (shashwathi) [inline testing/e2e/cluster_supply_chain_test.go:42]: include the actual output you received in the command as well for the error

File: testing/e2e/cluster_supply_chain_test.go [hunk 6]
```diff
@@ -29,14 +30,49 @@ import (
...
+					t.FailNow()
+				}
+				const expectedFooter = `
```
Comment (shashwathi) [inline testing/e2e/cluster_supply_chain_test.go:45]: change it to variable

File: testing/e2e/cluster_supply_chain_test.go [hunk 7]
```diff
@@ -29,14 +30,49 @@ import (
...
+					t.Errorf("Expected %s to be present in the output", expectedHeader)
+					t.FailNow()
+				}
+				match, err := regexp.MatchString(`.*.\n[ ]{3,}labels[ ]{3,}apps\.tanzu\.vmware\.com\/workload-type[ ]{3,}web\n.*`, output)
+				if err != nil {
+					t.Error("Error while validating the output", err)
+					t.FailNow()
+				}
+				if !match {
+					t.Error("Expected 'labels   apps.tanzu.vmware.com/workload-type   web' to be present in the output")
```
Comment (shashwathi) [inline testing/e2e/cluster_supply_chain_test.go:71]: include the actual output you received in the command as well for the error

**Review by shashwathi — COMMENTED** [review]
Looks good. Minor comments.

File: testing/e2e/cluster_supply_chain_test.go (hunk 5 above)
Comment (odinnordico) [reply testing/e2e/cluster_supply_chain_test.go]: Added to the error message

File: testing/e2e/cluster_supply_chain_test.go (hunk 7 above)
Comment (odinnordico) [reply testing/e2e/cluster_supply_chain_test.go]: Added output to the error message

File: testing/e2e/cluster_supply_chain_test.go (hunk 6 above)
Comment (odinnordico) [reply testing/e2e/cluster_supply_chain_test.go]: Done

File: testing/suite/integration.go [hunk 8]
```diff
@@ -143,6 +147,20 @@ func (cl CommandLineIntegrationTestCase) Run(t *testing.T, ctx context.Context,
 		if cl.Skip {
 			t.SkipNow()
 		}
+		if len(cl.RequireEnvs) > 0 {
```
Comment (shashwathi) [inline testing/suite/integration.go:150]: you can remove this if condition

**Review by shashwathi — APPROVED** [review]

File: testing/suite/integration.go (hunk 8 above)
Comment (odinnordico) [reply testing/suite/integration.go]: removed

**Review by shashwathi — APPROVED** [review]

**Pull request merged**